File: startswith.py
#!/usr/bin/python
import sys
import re

if len(sys.argv) < 2:
   print "Usage: cat shakespeare.txt | startswith.py searchterm \n"
   sys.exit() 

term = sys.argv[1]

startswith = re.compile("^" + term)
nonletters = re.compile("[^a-z A-Z]")

for line in sys.stdin:
  line = line[:-1] # remove '\n'
  line = nonletters.sub("", line)
  words = line.split()
  for word in words:
     if startswith.match(word):
        print word

File: startswith.pl
#!/usr/bin/perl

$#ARGV == 0 or die "Usage: cat shakespeare.txt | startswith.pl searchterm \n";

$term = $ARGV[0];

while (<STDIN>){
  chop;
  $line = $_;
  $_ =~ s/[^a-zA-Z ]//g;  # only letters and spaces remain after this.
  @words = split;
  for ($i=0; $i <= $#words; $i++){
      if ( $words[$i] =~ m/^($term)/ ){
         print "$line\n"; # to STDOUT
      }
  }
}
